SU2_CFD/src/output/filewriter/CCGNSFileWriter.cpp

```diff
@@ -0,0 +1,306 @@
+
+  SU2_MPI::Allreduce(&nLocalVertex, &MaxLocalVertex, 1,
+                       MPI_UNSIGNED_LONG, MPI_MAX, MPI_COMM_WORLD);
+  SU2_MPI::Gather(&Buffer_Send_nVertex, 1, MPI_UNSIGNED_LONG,
+                    Buffer_Recv_nVertex,  1, MPI_UNSIGNED_LONG,
+                    MASTER_NODE, MPI_COMM_WORLD);
+
+  const vector<string> fieldNames = dataSorter->GetFieldNames();
+  unsigned short nVar = fieldNames.size();
+
+  su2double *buffLocData = new su2double[MaxLocalVertex * nVar]();
+  su2double *buffGlobData = nullptr;
+
+  index = 0;
+  for (int iPoint = 0; iPoint < nLocalVertex; iPoint++) {
+      for(int varCount = 0; varCount < nVar; varCount++){
+          buffLocData[index] = dataSorter->GetData(varCount, iPoint);
+          index++;
+      }
+  }
+
+  #ifdef HAVE_MPI
+      SU2_MPI::Barrier(MPI_COMM_WORLD);
+  #endif
+
+  if (rank == MASTER_NODE)
+      buffGlobData = new su2double[nProcessor * nVar * MaxLocalVertex]();
+
+  SU2_MPI::Gather(buffLocData, (int)MaxLocalVertex*nVar, MPI_DOUBLE,
+                  buffGlobData, (int)MaxLocalVertex*nVar, MPI_DOUBLE,
+                  MASTER_NODE, MPI_COMM_WORLD);
+
+  su2double **GlobalData;
+  if (rank == MASTER_NODE){
+      GlobalData = new su2double*[nVar];
+      for(int i = 0; i < nVar; i++)
+          GlobalData[i] = new su2double[dataSorter->GetnPointsGlobal()];
+
+    index = 0;
+    int index2 = 0;
+    for (iProcessor = 0; iProcessor < nProcessor; iProcessor++) {
+        for(int j = 0; j < Buffer_Recv_nVertex[iProcessor]; j++){
+            index  = (iProcessor*MaxLocalVertex*fieldNames.size() +
+                                              j*fieldNames.size());
+            for (int i = 0; i < nVar; i++)
+                GlobalData[i][j+index2] = buffGlobData[index + i];
+        }
+        index2 += Buffer_Recv_nVertex[iProcessor];
+    }
+//    delete [] Buffer_Recv_nVertex;
+//    delete [] buffGlobData;
+  }
+// delete [] buffLocData;
+
+// Free memory: *Buffer_Recv_nVertex(solo master), *buffGlobData(solo master), *buffLocData
+  #ifdef HAVE_MPI
+      SU2_MPI::Barrier(MPI_COMM_WORLD);
+  #endif
+
+  if (rank == MASTER_NODE){
+  #ifdef HAVE_CGNS
+
+    /*---    Internal Variables    ---*/
+    int cgns_err, cgns_file, cgns_file1, cgns_flow, cgns_field, nbndry, NormalIndex, bc_wall, bc_far, bc_nulli, coord_index;
+    int nbases, base_number, cell_dim, phys_dim, nzones, zone_number, ncoords, nsections, i, S;
+    int bcnum, parent_flag;
+    cgsize_t isize[3][1];
+    cgsize_t start, end, ElementDataSize, NormalListSize, npnts, j, k, z, p, q, NewElementDataSize, dimensions;
+    cgsize_t *ConnectOffset;
+    cgsize_t pnts[1][2];
+    cgsize_t A[4], B[4];
+    cgsize_t *Elements = NULL;
+    cgsize_t *NewElements = NULL;
+    PointSetType_t ptset_type;
+    BCType_t bocotype;
+    DataType_t datatype;
+    GridLocation_t location;
+    ElementType_t type;
+    DataClass_t dataclass;
+    #define MIXED 20
```

Comment:
It would be preferable to use a variable rather than a macro for these constants.
(because macros maybe defined somewhere else)

Reply:
I changed it into a variable.